app/pages/lab/workflow.jsx

```diff
@@ -0,0 +1,1042 @@
+    if (this.canUseTask(this.props.project, "transcription-task") && usingTranscriptionTask) {
+      const env = process.env.NODE_ENV;
+      return `https://fe-project.zooniverse.org/projects/${owner}/${name}/classify/workflow/${this.props.workflow.id}?env=${env}`;
+    } else {
+      const viewQuery = {workflow: this.props.workflow.id, reload: this.state.forceReloader};
+      return this.context.router.createHref({
+        pathname: `/projects/${owner}/${name}/classify`,
+        query: viewQuery
+      });
+    }
+  }
+
+  showCreateWorkflow() {
+    return this.setState({workflowCreationInProgress: true});
+  }
+
+  hideCreateWorkflow() {
+    return this.setState({workflowCreationInProgress: false});
+  }
+
+  handleWorkflowCreation(workflow) {
+    this.hideCreateWorkflow();
+    const newLocation = Object.assign({}, this.props.location, {pathname: `/lab/${this.props.project.id}/workflows/${workflow.id}`});
+    this.context.router.push(newLocation);
+    this.props.project.uncacheLink('workflows');
+    return this.props.project.uncacheLink('subject_sets');
+  } // An "expert" subject set is automatically created with each workflow.
+
+  canUseTask(project, task) {
+    return Array.from(project.experimental_tools).includes(task);
+  }
+
+  handleTaskChange(taskKey, taskDescription) {
+    const changes = {};
+    changes[`tasks.${taskKey}`] = taskDescription;
+    return this.props.workflow.update(changes).save();
+  }
+
+  isThereNotADefinedTask() {
+    const workflowTasks = Object.keys(this.props.workflow.tasks);
+    return workflowTasks.length === 0;
+  }
+
+  showTaskAddButtons() {
+    return this.setState(prevState => ({ showTaskAddButtons: !prevState.showTaskAddButtons }));
+  }
+
+  render() {
+    let definition;
+    window.editingWorkflow = this.props.workflow;
+
+    const noTasksDefined = Object.keys(this.props.workflow.tasks).length === 0;
+
+    const stats_completeness_type = this.props.workflow.configuration.stats_completeness_type != null ? this.props.workflow.configuration.stats_completeness_type : 'retirement';
+
+    const projectLiveWorkflowActive = this.props.project.live && this.props.workflow.active;
+    const projectLiveWorkflowInactive = this.props.project.live && !this.props.workflow.active;
+    const disabledIfLive = classnames({ 'disabled-section': projectLiveWorkflowActive });
+    const disabledIfWorkflowInactive = classnames({ 'disabled-section': projectLiveWorkflowInactive });
+    const taskEditorClasses = classnames({'column': true, 'disabled-section': projectLiveWorkflowActive});
+
+    return (
+      <div className="edit-workflow-page">
+        <h3>{this.props.workflow.display_name} #{this.props.workflow.id}{' '}
+          <button onClick={this.showCreateWorkflow.bind(this)} disabled={this.state.workflowCreationInProgress} title="Copy workflow">
```

Comment:
`this.showCreateWorkflow` should be declared in the constructor, following the pattern here:
https://github.com/zooniverse/Panoptes-Front-End/blob/6b128109d64c4d30429e37f44dc62e3f6fa3ed64/app/classifier/tasks/highlighter/index.jsx#L47-L52

Reply:
CoffeeScript did this for us, with `@`, but that’s been removed in this PR. Now, new event handlers are generated on every render. Each time you toggle a checkbox, or type in a text box, a new set of event handlers is added to the component. That leads to memory leaks in the browser.

Reply:
I can go through the code and make this change as review suggestions, or open a PR. Either way, it’s not a lot of work.

Reply:
Since this changes the behaviour of the component from the CoffeeScript version, I’d prefer this change to be made in this PR, not after.

Reply:
#6225.

Reply:
Fixed now, so we have parity with the old `createReactClass` code.